hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/CreateTranslator.java

```diff
@@ -27,200 +27,268 @@
                 decoder::decodeCreateFungibleTokenWithMetadataAndCustomFees);
-        createSelectorsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V1, decoder::decodeCreateNonFungibleV1);
-        createSelectorsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V2, decoder::decodeCreateNonFungibleV2);
-        createSelectorsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V3, decoder::decodeCreateNonFungibleV3);
-        createSelectorsMap.put(CREATE_NON_FUNGIBLE_TOKEN_WITH_METADATA, decoder::decodeCreateNonFungibleWithMetadata);
-        createSelectorsMap.put(
+        createMethodsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V1, decoder::decodeCreateNonFungibleV1);
+        createMethodsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V2, decoder::decodeCreateNonFungibleV2);
+        createMethodsMap.put(CREATE_NON_FUNGIBLE_TOKEN_V3, decoder::decodeCreateNonFungibleV3);
+        createMethodsMap.put(CREATE_NON_FUNGIBLE_TOKEN_WITH_METADATA, decoder::decodeCreateNonFungibleWithMetadata);
+        createMethodsMap.put(
                 CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V1, decoder::decodeCreateNonFungibleWithCustomFeesV1);
-        createSelectorsMap.put(
+        createMethodsMap.put(
                 CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V2, decoder::decodeCreateNonFungibleWithCustomFeesV2);
-        createSelectorsMap.put(
+        createMethodsMap.put(
                 CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V3, decoder::decodeCreateNonFungibleWithCustomFeesV3);
-        createSelectorsMap.put(
+        createMethodsMap.put(
                 CREATE_NON_FUNGIBLE_TOKEN_WITH_METADATA_AND_CUSTOM_FEES,
                 decoder::decodeCreateNonFungibleWithMetadataAndCustomFees);
     }
 
     @Override
-    public boolean matches(@NonNull HtsCallAttempt attempt) {
+    public @NonNull Optional<SystemContractMethod> identifyMethod(@NonNull HtsCallAttempt attempt) {
+        requireNonNull(attempt);
         final var metaConfigEnabled =
                 attempt.configuration().getConfigData(ContractsConfig.class).metadataKeyAndFieldEnabled();
-        final var metaSelectors = Set.of(
-                CREATE_FUNGIBLE_TOKEN_WITH_METADATA,
-                CREATE_FUNGIBLE_TOKEN_WITH_METADATA_AND_CUSTOM_FEES,
-                CREATE_NON_FUNGIBLE_TOKEN_WITH_METADATA,
-                CREATE_NON_FUNGIBLE_TOKEN_WITH_METADATA_AND_CUSTOM_FEES);
-        return createSelectorsMap.keySet().stream()
-                .anyMatch(selector -> metaSelectors.contains(selector)
-                        ? attempt.isSelectorIfConfigEnabled(metaConfigEnabled, selector)
-                        : attempt.isSelector(selector));
+
+        for (final var method : createMethodsMap.keySet()) {
+            final var isMetadataMethod = method.hasVariant(Variant.WITH_METADATA);
+            Optional<SystemContractMethod> m = isMetadataMethod
+                    ? metaConfigEnabled ? attempt.isMethod(method) : Optional.empty()
+                    : attempt.isMethod(method);
+            if (m.isPresent()) return m;
```

Comment:
`?:` inside of `?:` is pretty hard to read here

Reply:
Nuts.  This is exactly the kind of thing I mention when doing code reviews for _other_ people.